presto-main/src/main/java/com/facebook/presto/sql/relational/SqlFunctionUtils.java

```diff
@@ -108,6 +108,9 @@ public static RowExpression getSqlFunctionRowExpression(
                         Optional.empty(),
                         sqlFunctionProperties,
                         sessionFunctions,
+                        // TODO: use session to determine if this is a native query
```

Comment:
Why can't we address this TODO now?

Reply:
Detailed conversation in this thread: https://github.com/prestodb/presto/pull/19791/files/189afe0a571d7ee128f242fd803a6098d967861b#r1243767843

In short,
> The SqlFunctionUtils, which are used in the sql function code path does not currently properly support native, since there is no access to the session there, only connector session.

Reply:
the solution would be to add the isNative flag to the connector session and pass it through when we create the connector session.  I think that's a very reasonable to do, and there may be other places that would want to handle things differently depending on the native flag too, so could be generally useful.